senders/test_server_test.go

```diff
@@ -0,0 +1,73 @@
+package senders
+
+import (
+	"bufio"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"net/http"
+)
+
+func newTestServer(port int) *testServer {
+	metricLines := make([]string, 0)
```

Comment:
You can shorten this function to just be
```
return &testServer{port: port}
```
You can append to a nil slice and the right things will happen.